build.sh

```diff
@@ -3,28 +3,101 @@
 # Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 #-------------------------------------------------------------------------------------------------------
 
-if [ ! -d "BuildLinux" ]; then
-    mkdir BuildLinux;
-fi
+SAFE_RUN() {
+    SF_RETURN_VALUE=$($1 2>&1)
 
-pushd BuildLinux > /dev/null
+    if [[ $? != 0 ]]
+    then
+        echo $SF_RETURN_VALUE
+        exit
+    fi
+}
+
+PRINT_USAGE() {
+    echo ""
+    echo "build.sh [options]"
+    echo ""
+    echo "options:"
+    echo "--help, -h : Show help"
+    echo "--debug, -d : Debug build"
+    echo "CXX=<path to clang++>"
+    echo "CC=<path to clang>"
+    echo ""
+    echo "example: ./build.sh CXX=/path/to/clang++ CC=/path/to/clang"
+}
 
+_CXX=""
+_CC=""
 DEBUG_BUILD=0
-while getopts ":d" opt; do
-    case $opt in
-        d)
+
+while test $# -gt 0
+do
+    if [[ "$1" =~ "CXX=" ]]; then
+        _CXX=$1
+        _CXX=${_CXX:4}
+    fi
+
+    if [[ "$1" =~ "CC=" ]]; then
+        _CC=$1
+        _CC=${_CC:3}
+    fi
+
+    if [[ "$1" == "--help" || "$1" == "-h" ]]; then
+        PRINT_USAGE
+        exit
+    fi
+    
```

Comment:
> nit: trim trailing whitespace in this line
turned of automatic whitespace trimming on all the editors. there we have this. I'm still +100 on merging that whitespace fix PR though.